.github/workflows/ibis-backends.yml

```diff
@@ -704,6 +704,70 @@ jobs:
         with:
           flags: backend,${{ matrix.backend.name }},${{ runner.os }},python-${{ steps.install_python.outputs.python-version }}
 
+  test_flink:
+    name: Flink ${{ matrix.os }} python-${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        python-version:
+          - "3.9"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - uses: extractions/setup-just@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: download backend data
+        run: just download-data
+
+      - name: start services
+        run: docker compose up --wait flink
+
+      - name: install python
+        uses: actions/setup-python@v4
+        id: install_python
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: install poetry
+        run: python -m pip install --upgrade pip 'poetry==1.6.1'
+
+      - name: install ibis
+        run: poetry install --without dev --without docs --extras flink
+
+      # FIXME(deepyaman): We reinstall pandas~=1.5 to get a version with
+      #   ArrowDtype, but this step will be removed once flink relaxes
+      #   its overly-restrictive dependencies (specifying pandas<1.4.0).
+      - name: install ibis-flink deps
+        run: poetry run pip install apache-flink grpcio-status pytest-split pandas~=1.5
```

Comment:
This can be simplified now; see https://github.com/ibis-project/ibis/pull/7542

Reply:
lgtm, how about I rebase the code on yours after that PR is merged it?

Reply:
Sounds good, let me ask for a quick review

Reply:
#7542 is merged and this can be rebased now!